src/main/java/org/testng/annotations/AfterMethod.java

```diff
@@ -40,6 +40,16 @@
    */
   public String[] dependsOnMethods() default {};
 
+  /**
+   * Causes this method to be invoked only if the test method belongs to a listed group.
+   * It can be used if different cleanups are needed for different groups.  Omitting
+   * this or setting it to an empty list will cause this method to run after every
+   * test method, regardless of which group it belongs to.  Otherwise, this method
+   * is only invoked if the test method that was invoked belongs to one of the groups
+   * listed here.
+   */
```

Comment:
I'm not clear what the difference is between this and the `groups` attribute, could you clarify this in the comment?

Reply:
This part is exactly about it: “Omitting this or setting it to an empty list will cause this method to run after every test method, regardless of which group it belongs to”—this is what happens if you just set the `groups` attribute. A better wording is beyond my English, but I've also explained this in the comment for the `groups` attribute now (see my amended/rebased commit), similar to `@BeforeMethod`. I think that's a better place to explain the difference because that's where people start looking when they need “a method that runs before every method from some group”, which is exactly what my new attribute does and the `groups` attribute doesn't.